src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    internal static class MakeLocalFunctionStaticHelper
+    {
+        public static bool IsStaticLocalFunctionSupported(SyntaxTree tree)
+            => tree.Options is CSharpParseOptions csharpOption && csharpOption.LanguageVersion >= LanguageVersion.CSharp8;
+
+        public static bool TryGetCaputuredSymbols(LocalFunctionStatementSyntax localFunction, SemanticModel semanticModel, out ImmutableArray<ISymbol> captures)
+        {
+            var dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
+            captures = dataFlow.CapturedInside;
+
+            return dataFlow.Succeeded;
+        }
+
+        public static async Task<Document> MakeLocalFunctionStaticAsync(
+            LocalFunctionStatementSyntax localFunction,
+            ImmutableArray<ISymbol> captures,
+            SemanticModel semanticModel,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var root = (await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false))!;
+
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localFunction, cancellationToken);
+            var parameters = CreateParameterSymbols(captures);
+
+            // Finds all the call sites of the local function
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(
+                localFunctionSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            // Now we need to find all the refereces to the local function that we might need to fix.
+            var shouldWarn = false;
+            var invocationReferences = new List<InvocationExpressionSyntax>();
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                foreach (var location in referencedSymbol.Locations)
+                {
+                    // Since this is a local function, all reference must be in the same tree.
```

Comment:
i agree.... however, just do a test for this so you don't crash if you're wrong in the future :)

Reply:
I have changed the find reference scope to the single document, so this should be fine now :)